// @sentry/sveltekit/build/cjs/vite/recastTypescriptParser.js
Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const parser$1 = require('@babel/parser');

const parser = {
  parse: (source) =>
    parser$1.parse(source, {
      strictMode: false,
      allowImportExportEverywhere: true,
      allowReturnOutsideFunction: true,
      startLine: 1,
      tokens: true,
      plugins: [
        'typescript',
        'asyncGenerators',
        'bigInt',
        'classPrivateMethods',
        'classPrivateProperties',
        'classProperties',
        'classStaticBlock',
        'decimal',
        'decorators-legacy',
        'doExpressions',
        'dynamicImport',
        'exportDefaultFrom',
        'exportNamespaceFrom',
        'functionBind',
        'functionSent',
        'importAssertions',
        'exportExtensions' ,
        'importMeta',
        'nullishCoalescingOperator',
        'numericSeparator',
        'objectRestSpread',
        'optionalCatchBinding',
        'optionalChaining',
        [
          'pipelineOperator',
          {
            proposal: 'minimal',
          },
        ],
        [
          'recordAndTuple',
          {
            syntaxType: 'hash',
          },
        ],
        'throwExpressions',
        'topLevelAwait',
        'v8intrinsic',
      ],
      sourceType: 'module',
    }),
};

exports.parser = parser;
